controllers/helmchartproxy/helmchartproxy_controller.go

```diff
@@ -218,6 +218,12 @@ func (r *HelmChartProxyReconciler) reconcileNormal(ctx context.Context, helmChar
 			continue
 		}
 
+		// Don't reconcile if the Cluster is paused.
+		if cluster.Spec.Paused {
```

Comment:
```suggestion
		if annotations.IsPaused(cluster, cluster) {
```
Seems like the convention in CAPI is to use this annotation getter.

Reply:
Ah good tip! Just a small adjustment to check the paused annotation on the `helmChartProxy` and the `cluster.Spec.Paused` via the call you suggest. I'm also going to move this into `reconcileForCluster` which I think makes more sense.